include/zephyr/dt-bindings/input/input-event-codes.h

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DT_BINDINGS_INPUT_INPUT_EVENT_CODES_H_
+#define ZEPHYR_INCLUDE_DT_BINDINGS_INPUT_INPUT_EVENT_CODES_H_
+
+/**
+ * @defgroup input_events Input Event Definitions
+ * @ingroup input_interface
+ * @{
+ */
+
+
+/**
+ * @name Input event types.
+ * @anchor INPUT_EV_CODES
+ * @{
+ */
+#define INPUT_EV_KEY 0x01
+#define INPUT_EV_REL 0x02
+#define INPUT_EV_ABS 0x03
+#define INPUT_EV_MSC 0x04
+#define INPUT_EV_VENDOR_START 0xf0
+#define INPUT_EV_VENDOR_STOP 0xff
+/** @} */
+
+/**
+ * @name Input event KEY codes.
+ * @anchor INPUT_KEY_CODES
+ * @{
+ */
```

Comment:
A reference as to where these code originate (since they are not just an enumeration starting from 0) would be nice.

Reply:
Yeah good point, added a link at the top of the file and a note asking to use the same codes if codes are available in linux as well.